src/core/utils/configutils.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * Used to parse config options, defaulting to different synonyms and
+ * finally a default value. Option names with periods will be parsed
+ * as multiple child object accessors, i.e. trying to access 'first.second.option'
+ * will first look for config['first']['second']['option'].
+ *
+ * This is mostly needed for boolean config values, since boolean operators,
+ * which we commonly use for defaulting config options, do not work properly
+ * in those cases.
+ * @param {Object} topLevelConfig
+ * @param  {Array<string>}
+ * @param {any} defaultValue
+ */
+export function parseConfig (topLevelConfig, synonyms, defaultValue) {
```

Comment:
Let's give this a more descriptive name. Maybe "defaultConfigOption" or something like that. The name "parseConfig" indicates this parses the entire config object and returns some parsed version of it. I think we should rename topLevelConfig to be just "config". Theoretically, someone could use at any level of config.

Reply:
sure I definitely had a lot of trouble coming up with accurate and descriptive names for this, updated! there's also already a parseConfig in answers-umd so its good not to share names